exporter/datasetexporter/factory.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datasetexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datasetexporter"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datasetexporter/internal/metadata"
+)
+
+const CfgTypeStr = "dataset"
+
+// NewFactory created new factory with DataSet exporters.
+func NewFactory() exporter.Factory {
+	return exporter.NewFactory(
+		CfgTypeStr,
+		createDefaultConfig,
+		exporter.WithLogs(createLogsExporter, metadata.Stability),
+		exporter.WithMetrics(createMetricsExporter, metadata.Stability),
```

Comment:
The issue and readme only mention logs. Is the intention to support metrics and traces as well?

Reply:
For now I have follow up PR, where I am supporting logs and traces - https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20870. 

I can adjust it in this PR or I can do it in the real PR.

Reply:
I think it makes sense to add the implementation in that PR, I just checking since the metadata file and linked issue only mentioned logs. So long as this is in sync with what is in the metadata file [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20733/files#diff-15048bd70a8a5544947470c6c938c9e0d95e3a2aca2e18a74c24b7eede632770R6) I think we're good.

Reply:
I have removed metrics in this PR and updated the metadata, so it should be in sync.